### ISSUE TITLE
Log and describe don't work correctly

### ISSUE THREAD

**Issue by mariusGundersen** [cake-contrib/Cake_Git#85]
There is a [simple command in git](https://stackoverflow.com/questions/12082981/get-all-git-commits-since-last-tag) to get every commit since previous tags, something that is useful when creating release notes:

    git log $(git describe --tags --abbrev=0)..HEAD --oneline

The equivalent code for Cake.Git would be

    GitLog(rootDir, GitDescribe(rootDir,  GitDescribeStrategy.Tags));

Unfortunately this does not produce the same result. From the source code it looks like it uses a simple linq expression that stops when it finds the tag, while the [libgit2sharp documentation recommends a different way to do it](https://github.com/libgit2/libgit2sharp/wiki/git-log). For example, given the following git history:

```
        * - HEAD - merge
        |\
        | * - commit1
        | |
        | * - commit2
 Tag2 - * |
        |/
        * - Tag1
        |
```

The command line snippet above lists three lines, the `merge`, `commit1` and `commit2`, but Cake.Git only lists the `merge` commit, since it stops when the linq expression reaches the first tag. 

If Cake.Git instead use `ExcludeReachableFrom` as described in the libgit2sharp documentation it would produce the correct output.

**Comment by mariusGundersen** [comment]
Ok, having looked into this I've found that the correct way to do this is as follows:

```csharp
GitLog(rootDir, GitDescribe(rootDir,  false, GitDescribeStrategy.Tags, 0));
```

This will find the correct number of commits, but it's limited in that libgit2sharp will only take the 10 most recent tags into consideration, so it might miss some: https://github.com/libgit2/libgit2sharp/blob/321ac068099839d7a428398e6168b4c59f338fb9/LibGit2Sharp/Core/Proxy.cs#L762

**Comment by mariusGundersen** [comment]
Ok, I thought this would work, but I'm reopening it again because it didn't. The code in GitLog only works when the id is for a commit, so it doesn't work for a tag. 

There are a few different ways to solve this, and I'm not sure which one you would prefer. Either GitDescribe has to be changed/overloaded to return the hash of the commit the tag points to, or GitLog has to be changed/overloaded to accept a tag so that it can look up the hash.

**Comment by devlead** [comment]
fixed by #86
